### ISSUE TITLE
RC Switch binding to control RF transmitters

### ISSUE THREAD

**Issue by dominicdesu** [openhab/openhab#1993]
As far as I know there are currently two ways in openHAB to control 433 Mhz RC Power Switches with a RF transmitter connected to the GPIO Pins of a raspberryPi:
1. Exec Binding: Installing rc-switch and wiringPi and calling them via the exec binding
2. Pilight binding: Installing pilight service and call the service via pilight binding

Both approaches require external software to be installed and configured. Would be much nicer and easier if this is all intergrated within a single openHAB binding.
This should be possible with a java library called [Pi-RC-Switch](https://github.com/entrusc/Pi-RC-Switch) (which heavily uses the [Pi4J library](http://pi4j.com)).

**Comment by mroeckl** [comment]
Some time ago I implemented a prototype for this binding which embeds Pi-RC-Switch and Pi4J because I did not want to rely on external libraries as you described. If this binding is interesting for the openHAB community, I would be glad to contribute it to the project.

**Comment by dominicdesu** [comment]
You can probably guess my reply as I created this issue but just in case: That would be awesome! :-)

**Comment by mroeckl** [comment]
I asked in the OpenHAB Google group if the community is interested to add this binding as a new feature ([here](https://groups.google.com/forum/#!category-topic/openhab/features/_TPI5STjP8s)) but so far nobody replied. Seems that we are the only potential users. Thus, I committed the code on my github:  [mroeckl/openhab-rcswitch](https://github.com/mroeckl/openhab-rcswitch). Let me know if there are any problems.

**Comment by dominicdesu** [comment]
Hi,
as you've already written the complete binding, why don't you commit it to OpenHAB? Even though nobody answered in the google group there might still be interest in the future. Especially as this binding makes it really easy to connect RF Switches with openHAB!
(And if you're lucky, you even get a thorough code review and can learn from it.)

About your second question in the group: As far as I know, bundles should always be independent of each other... but I am not an expert. Why don't you commit it with the Pi4J/RCSwitch library and if somebody thinks this should be generalized a new ticket can be created.
Maybe you can import the RCSwitch as a jar library instead of the source code? I am not sure what the general recommendations (Library or source code) are in this case, however.

Anyway, thanks a lot for you effort! Looking forward to use the binding!

**Comment by dominicdesu** [comment]
I was just looking at the configuration description (README.md).
The gpio PIN should be always the same, while the group address and device address might change per device, right?
Maybe it would make sense to keep the gpio PIN in the openhab.cfg but put the other settings directly into the item configuration, like this:
Switch  SleepingRoom    {rcswitch="{group=11111, device=1}"}
or even
Switch  SleepingRoom    {rcswitch="11111:1"}

This would be similar to other bindings like [hue](https://github.com/openhab/openhab/wiki/Hue-Binding) or [enocean](https://github.com/openhab/openhab/wiki/EnOcean-Binding). What do you think?

**Comment by mroeckl** [comment]
Thanks for your feedback, dominicdesu. I looked at the other bindings and re-factored some code which makes the binding now much simpler.
I agree to your recommendations. Being able to add the group and device address in the item configuration makes it much easier. I committed the new code to my github. I will prepare a pull request for openhHAB to get this binding integrated into the official release.
Thank you very much for your support.
